api/src/wfl/module/covid.clj

```diff
@@ -27,6 +28,37 @@
 (defn start-covid-workload!
   [])
 
+(defn ^:private snapshot-imported?
+  "Check if snapshot SNAPSHOT_ID in TDR has been imported to WORKSPACE
+  as SNAPSHOT_REFERENCE_ID and can be successfully fetched."
+  [{:keys [snapshot_id snapshot_reference_id] :as _source_details}
+   {:keys [workspace] :as _executor}]
+  (letfn [(fetch [] (rawls/get-snapshot-reference workspace
+                                                  snapshot_reference_id))]
+    (some? (and snapshot_id snapshot_reference_id (util/do-or-nil (fetch))))))
```

Comment:
Thanks @tbl3rd for the lesson on `some?`s.

Reply:
I'm not sure why this needs to do a null check on the snapshot_id. Furthermore, this doesn't check if that snapshot has been imported as that snapshot reference. It just checks if they're both non-nil and that the snapshot reference exists.
Maybe you can just check if the snapshot reference exists?

Reply:
@tbl3rd (summary from our Slack conversation): I like this suggestion and am implementing it.  It communicates the same information without throwing out the snapshot reference.

@ehigham yeah, you're right that the `snapshot_id` check is unneeded, I will remove.  Are you okay with me swallowing any error thrown by the fetch in this case?